### ISSUE TITLE
[Link] Update a11y guidance for Link groups

### ISSUE THREAD

**Issue by alina-jacob** [carbon-design-system/carbon-website#4932]
## Acceptance criteria

In this issue https://github.com/carbon-design-system/ibm-products/issues/9571, we have identified an a11y violation that links should have a minimum spacing of 12px between each other. This information needs to be updated for link.
**To-do:**
- [ ] Update the [Interaction guidance](https://carbondesignsystem.com/components/link/usage/#interactions) or [Accessibility guidance](https://carbondesignsystem.com/components/link/accessibility/) for Link groups. 
- [ ] Worth doing a quick audit to identify components where interaction and grouping is common. Noting sets that come to mind:
       - Tag groups
       -

**Comment by embrown-ibm** [comment]
@alina-jacob Is there an estimated time as to when this will be resolved?   We missed getting this fixed for the most recent GA of one of my products.  That product has an upcoming fixpack to be delivered and I have another product to GA soon.  We would like to get this resolved in both of those deliverables.  Thanx!!!
